subsys/bluetooth/audio/bap_broadcast_sink.c

```diff
@@ -566,211 +566,73 @@ static bool codec_lookup_id(const struct bt_pacs_cap *cap, void *user_data)
 	return true;
 }
 
-struct store_base_info_data {
-	struct bt_bap_broadcast_sink_bis bis[CONFIG_BT_BAP_BROADCAST_SNK_STREAM_COUNT];
-	struct bt_bap_broadcast_sink_subgroup subgroups[CONFIG_BT_BAP_BROADCAST_SNK_SUBGROUP_COUNT];
-	struct bt_audio_codec_cfg *subgroup_codec_cfg;
-	uint32_t valid_indexes_bitfield;
-	uint8_t subgroup_count;
-	uint8_t bis_count;
-};
-
-static bool merge_bis_and_subgroup_data_cb(struct bt_data *data, void *user_data)
+static bool base_subgroup_bis_count_cb(const struct bt_bap_base_subgroup *subgroup, void *user_data)
 {
-	struct bt_audio_codec_cfg *codec_cfg = user_data;
-	int err;
-
-	err = bt_audio_codec_cfg_set_val(codec_cfg, data->type, data->data, data->data_len);
-	if (err < 0) {
-		LOG_DBG("Failed to set type %u with len %u in codec_cfg: %d", data->type,
-			data->data_len, err);
+	uint8_t *bis_cnt = user_data;
+	int ret;
 
+	ret = bt_bap_base_get_subgroup_bis_count(subgroup);
+	if (ret < 0) {
 		return false;
 	}
 
+	*bis_cnt += (uint8_t)ret;
+
 	return true;
 }
 
-static bool base_subgroup_bis_index_cb(const struct bt_bap_base_subgroup_bis *bis, void *user_data)
+static int base_get_bis_count(const struct bt_bap_base *base)
```

Comment:
Instead of implementing this function I would suggest to either
1) Add that to the `bap_base.c` file and offer it as an API function
2) Use `bt_bap_base_get_bis_indexes` and count bits

Reply:
It was there before https://github.com/zephyrproject-rtos/zephyr/blob/main/subsys/bluetooth/audio/bap_broadcast_sink.c#L760-L773 

But could make sense - or even better: if possible to shuffle around in the code using it so it can be part of the iteration in `pa_decode_base`, picking up the subgroups and indexes?

Reply:
You are right - It's OK to keep as is then, although my suggestions still apply (perhaps for a future PR)